android/src/main/java/com/magicleap/magicscript/scene/nodes/dropdown/DropdownItemsListNode.kt

```diff
@@ -0,0 +1,159 @@
+                setNeedsRebuild(force = true)
+            }
+        }
+
+    var multiSelect = false
+
+    private val items = mutableListOf<UiDropdownListItemNode>()
+    private lateinit var itemsAdapter: DropdownItemsAdapter
+
+    override fun provideView(context: Context): View {
+        return LayoutInflater.from(context).inflate(R.layout.dropdown_list, null)
+    }
+
+    override fun provideDesiredSize(): Vector2 {
+        val tempAdapter = buildAdapter()
+        val size = calculateItemsSize(tempAdapter)
+        if (maxHeight != WRAP_CONTENT_DIMENSION && size.y > maxHeight) {
+            size.y = maxHeight
+        }
+        return size
+    }
+
+    override fun setupView() {
+        super.setupView()
+        val listView = view as ListView
+        itemsAdapter = buildAdapter()
+        listView.adapter = itemsAdapter
+        listView.setOnItemClickListener { parent, view, position, id ->
+            val item = items[position]
+            handleSelectionChangeRequest(item, !item.isSelected)
+        }
+    }
+
+    override fun onViewLoaded(viewRenderable: Renderable) {
+        super.onViewLoaded(viewRenderable)
+        viewRenderable.renderPriority = RENDER_PRIORITY
+    }
+
+    override fun addContent(child: TransformNode) {
+        super.addContent(child)
+        if (child is UiDropdownListItemNode) {
+            items.add(child)
+            if (child.selectRequested) {
+                handleSelectionChangeRequest(child, true)
+            }
+            child.onSelectionChangeRequest = { select ->
+                handleSelectionChangeRequest(child, select)
+            }
+            setNeedsRebuild(true)
+        }
+    }
+
+    override fun removeContent(child: TransformNode) {
+        super.removeContent(child)
+        if (child is UiDropdownListItemNode) {
+            items.add(child)
+            setNeedsRebuild(true)
+        }
+    }
+
+    // Calculates list view size based on list items size
+    private fun calculateItemsSize(itemsAdapter: DropdownItemsAdapter): Vector2 {
+        var width = 0f
+        var height = 0f
+        val listView = view as ListView
+        for (i in 0 until itemsAdapter.count) {
+            val childView = itemsAdapter.getView(i, null, listView)
+            val size =
+                childView.getSizeInMeters(context, WRAP_CONTENT_DIMENSION, WRAP_CONTENT_DIMENSION)
+            if (size.x > width) {
+                width = size.x
+            }
+            height += size.y
+        }
+        // height += Utils.pxToMeters(listView.dividerHeight * itemsAdapter.count, context)
+        return Vector2(width, height)
+    }
+
+    private fun buildAdapter(): DropdownItemsAdapter {
+        val adapter = DropdownItemsAdapter(context, R.layout.dropdown_item, items)
+        adapter.textSizePx = Utils.metersToFontPx(itemsTextSize, context)
+        return adapter
+    }
+
+    private fun handleSelectionChangeRequest(item: UiDropdownListItemNode, select: Boolean) {
```

Comment:
can you split this method? for example extract `handleMultiSelectChange`, `handleSelectChange` etc..